src/Symfony/Component/Security/Core/Security.php

```diff
@@ -69,4 +70,15 @@ public function getToken(): ?TokenInterface
     {
         return $this->container->get('security.token_storage')->getToken();
     }
+
+    /**
+     * Logout the current user automatically. Dispatch the logout event.
+     */
+    public function autoLogout(): void
+    {
+        $request = $this->container->get('request_stack')->getCurrentRequest();
+        $logoutEvent = new LogoutEvent($request, $this->container->get('security.token_storage')->getToken());
+        $this->container->get('event_dispatcher')->dispatch($logoutEvent);
```

Comment:
As the regular logout listeners are expected to return a response as in: https://github.com/symfony/symfony/blob/f963ae476a6534b29b638f47d19dc076a0afebd5/src/Symfony/Component/Security/Http/Firewall/LogoutListener.php#L121-L131

I wonder if it wouldn't be better to have this helper function also return the event's response? That way the caller can simply return that as response and be sure that the user is sent to the correct address without having to duplicate any URL/logic.

Reply:
Thanks, it is fixed.